controllers/models/index.js

```diff
@@ -40,16 +39,33 @@ exports.retrieveArticles = (id) => {
     //COUNT(*) counts all the items in the group
     //GROUP BY creates groups for all comments seperated by article_id
     //SQL hurts
-    let sqlQuery = `SELECT author, title, articles.article_id, topic, created_at, votes, article_img_url, COALESCE(comments.comment_count, 0) AS comment_count FROM articles
+    const sqlQuery = `SELECT author, title, articles.article_id, topic, created_at, votes, article_img_url, COALESCE(comments.comment_count, 0) AS comment_count FROM articles
     LEFT JOIN (
         SELECT comments.article_id, CAST(COUNT(*) AS INT) AS comment_count
         FROM comments
         GROUP BY comments.article_id
-    ) comments ON comments.article_id = articles.article_id`
-    sqlQuery += ` ORDER BY created_at DESC`
+    ) comments ON comments.article_id = articles.article_id ORDER BY created_at DESC`
     return retrieveData(sqlQuery)
 }
 
 exports.retrieveComments = (id) => {
     return db.query(`SELECT * FROM comments WHERE article_id=${id} ORDER BY created_at DESC`)
+}
+
+exports.postComment = (id, body) => {
+    return db.query(`
+    INSERT INTO comments (body, author, article_id, votes, created_at)
+    VALUES ($1, $2, ${id}, 0, '${new Date().toISOString()}')
+    RETURNING *
+    `, [body.body, body.username])
+    .catch((err) => {
+        console.log(err)
```

Comment:
We have another catch here in the model, a catch should always be at the end of our promise chain which in this case is in the controller and then passed to our middleware error handler using next. No need to throw custom errors unless psql isn't throwing an error for us and we want to identify an error. Any handling of a psql error code is the job of our error handling middleware.